electron/src/sso/SingleSignOn.ts

```diff
@@ -148,11 +148,17 @@ export class SingleSignOn {
         }
       };
 
-      session.protocol.registerStringProtocol(SingleSignOn.SSO_PROTOCOL, handleRequest, error => {
-        if (error) {
-          throw new Error(`Failed to register protocol. Error: ${error}`);
-        }
-      });
+      const isHandled = await new Promise<boolean>(resolve =>
+        session.protocol.isProtocolHandled(SingleSignOn.SSO_PROTOCOL, result => resolve(result as any)),
```

Comment:
The types state that the callback for `isProtocolHandled` returns an error but it actually returns a boolean.